addons/controllers/clusterbootstrap_controller.go

```diff
@@ -313,16 +611,17 @@ func (r *ClusterBootstrapReconciler) createOrPatchPackageInstallSecret(cluster *
 			log.Info("provider status does not have secretRef", "provider", provider)
 			return nil, nil
 		}
-		key := client.ObjectKey{Namespace: namespace, Name: secretName}
+		key := client.ObjectKey{Namespace: cluster.Namespace, Name: secretName}
 		if err := r.Get(r.context, key, secret); err != nil {
 			log.Error(err, "unable to fetch secret", "objectkey", key)
 			return nil, err
 		}
 	}
```

Comment:
Looks like we are not handling inline case now?

Reply:
Inline is captured in a separate issue: https://github.com/vmware-tanzu/tanzu-framework/issues/1694. 

The reasons of addressing it separately are:
- Inline is not a small change including the tests. This PR combines 2 issues already.
- We try to make the MVP in first, so that we could start testing the MVP logic in real env earlier.